tutorials/early_stopping/early_stopping.ipynb

```diff
@@ -0,0 +1,1067 @@
+    "        upper=1e-2,\n",
+    "        parameter_type=ParameterType.FLOAT,\n",
+    "        log_scale=True,\n",
+    "    ),\n",
+    "    RangeParameter(\n",
+    "        name=\"dropout\",\n",
+    "        lower=0.0,\n",
+    "        upper=0.5,\n",
+    "        parameter_type=ParameterType.FLOAT,\n",
+    "    ),\n",
+    "]\n",
+    "\n",
+    "search_space = SearchSpace(\n",
+    "    parameters=parameters,\n",
+    "    # NOTE: In practice, it may make sense to add a constraint\n",
+    "    # hidden_size_2 <= hidden_size_1\n",
+    "    parameter_constraints=[],\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "a8005e80",
+   "metadata": {},
+   "source": [
+    "## Setting up Metrics\n",
+    "\n",
+    "Ax has the concept of a Metric that defines properties of outcomes and how observations are obtained for these outcomes. This allows e.g. encodig how data is fetched from some distributed execution backend and post-processed before being passed as input to Ax.\n",
+    "\n",
+    "We will optimize the validation accuracy, which is a `TensorboardCurveMetric` that points to the logging directory assigned above. Note that we have set `is_available_while_running`, allowing for the metric to be queried as the trial progresses. This is critical for the early stopping strategy to monitor partial results."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "id": "0775a96e",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "class MyTensorboardMetric(TensorboardCurveMetric):\n",
+    "\n",
+    "    # NOTE: We need to tell the new Tensorboard metric how to get the id /\n",
+    "    # file handle for the tensorboard logs from a trial. In this case\n",
+    "    # our convention is to just save a separate file per trial in\n",
+    "    # the pre-specified log dir.\n",
+    "    @classmethod\n",
+    "    def get_ids_from_trials(cls, trials):\n",
+    "        return {\n",
+    "            trial.index: Path(log_dir).joinpath(str(trial.index)).as_posix()\n",
+    "            for trial in trials\n",
+    "        }\n",
+    "\n",
+    "    # This indicates whether the metric is queryable while the trial is\n",
+    "    # still running. This is required for early stopping to monitor the\n",
+    "    # progress of the running trial.ArithmeticError\n",
+    "    @classmethod\n",
+    "    def is_available_while_running(cls):\n",
+    "        return True"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "id": "a5c5a7d0",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "val_acc = MyTensorboardMetric(\n",
+    "    name=\"val_acc\",\n",
+    "    curve_name=\"val_acc\",\n",
+    "    lower_is_better=False,\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "d4f3ba5d",
+   "metadata": {},
+   "source": [
+    "## Setting up the OptimizationConfig\n",
+    "\n",
+    "The `OptimizationConfig` specifies the objective for Ax to optimize."
```

Comment:
Probably good to point to the fact that Ax also supports constraints and/or MOO. Though that might raise some questions about ES for MOO that we may want to work out first...